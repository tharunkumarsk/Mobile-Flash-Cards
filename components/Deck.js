import React, { Component } from "react";
import { StyleSheet, Text, View } from "react-native";
import Colors from "../constants/Colors";
import PropTypes from "prop-types";

export default class Deck extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    cardsCount: PropTypes.number.isRequired,
    cardId: PropTypes.number.isRequired
  };
  render() {
    const { title, cardsCount, cardId } = this.props;
    const cardStyle = cardId % 2 == 0 ? styles.redTile : styles.greenTile;
    return (
      <View style={[cardStyle, styles.MainContainer]}>
        <Text style={styles.cardViewTitle}> {title} </Text>
        <Text style={styles.cardViewCount}> {cardsCount} cards </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  MainContainer: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 10,
    height: 200
  },

  cardViewTitle: {
    fontSize: 40,
    color: "#000",
    textAlign: "center",
    marginTop: 20
  },
  cardViewCount: {
    fontSize: 20,
    color: "#000",
    textAlign: "center",
    marginBottom: 5
  },
  redTile: {
    backgroundColor: Colors.red
  },
  greenTile: {
    backgroundColor: Colors.green
  }
});
