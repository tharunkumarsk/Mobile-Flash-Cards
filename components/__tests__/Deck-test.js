import React from 'react';
import renderer from 'react-test-renderer';

import  Deck from '../Deck';

it(`renders correctly for odd numbers of cards`, () => {
  const tree = renderer.create(<Deck
    title = "Test"
    cardsCount = {2}
    cardId = {1}
  ></Deck>).toJSON();

  expect(tree).toMatchSnapshot();
});
it(`renders correctly for even numbers of cards`, () => {
  const tree = renderer.create(<Deck
    title = "Test"
    cardsCount = {2}
    cardId = {8}
  ></Deck>).toJSON();

  expect(tree).toMatchSnapshot();
});
