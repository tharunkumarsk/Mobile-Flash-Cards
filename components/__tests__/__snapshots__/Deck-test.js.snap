// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for even numbers of cards 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#b71845",
      },
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "flex": 1,
        "height": 200,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 40,
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
     
    Test
     
  </Text>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 20,
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
     
    2
     cards 
  </Text>
</View>
`;

exports[`renders correctly for odd numbers of cards 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#00965e",
      },
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "flex": 1,
        "height": 200,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 40,
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
     
    Test
     
  </Text>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 20,
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
     
    2
     cards 
  </Text>
</View>
`;
